exasol_script_languages_container_ci/lib/ci.py

```diff
@@ -52,15 +53,16 @@ def ci(ctx: click.Context,
     rebuild = False
     push_to_public_cache = False
 
-    IS_REBUILD = re.compile(r"refs/heads/rebuild/.*")
-    IS_MASTER = re.compile(r"refs/heads/master")
+    IS_REBUILD_REGEX = re.compile(r"refs/heads/rebuild/.*")
+    IS_MASTER_REGEX = re.compile(r"refs/heads/(master|main)")
 
-    rebuild = bool(IS_REBUILD.match(branch_name) or IS_MASTER.match(branch_name))
-    push_to_public_cache = bool(IS_MASTER.match(branch_name))
+    rebuild = bool(IS_REBUILD_REGEX.match(branch_name) or IS_MASTER_REGEX.match(branch_name))
+    push_to_public_cache = bool(IS_MASTER_REGEX.match(branch_name))
+    check_skip_test = not bool(IS_MASTER_REGEX.match(branch_name))
 
     flavor_path = (f"flavors/{flavor}",)
 
-    need_to_run = rebuild or check_if_need_to_build(config_file, flavor)
+    need_to_run = rebuild or check_if_need_to_build(config_file, flavor, git_access)
 
```

Comment:
I think, we should extract the evaluation of need_run and rebuild and all parts of them to a function, such that we can easily test it, because the logic gets complicated